src/Sarif/Baseline/ResultMatching/DataStructures/MatchedResults.cs

```diff
@@ -75,6 +75,8 @@ public Result CalculateBasedlinedResult(DictionaryMergeBehavior propertyBagMerge
             out Dictionary<string, object> originalResultMatchingProperties)
         {
             Result result = PreviousResult.Result.DeepClone();
+            result.Guid = result.Guid ?? Guid.NewGuid().ToString(SarifConstants.GuidFormat);
+            result.CorrelationGuid = result.CorrelationGuid ?? result.Guid;
             result.BaselineState = BaselineState.Absent;
```

Comment:
All of this code could be refactored into a helper, which is passed a result and baseline state, which clones the input results, sets the guids, sets the baseline state and returns the result. This helper could be used for all three places you changed in this file. Just a suggestion.

Reply:
Good idea. Fixed.